### PR TITLE
feat(data): add sheet addressing

### LINKED ISSUE #118: support addressing excel sheets
**Is your feature request related to a problem? Please describe.**
currently it is not possible to address a specific sheet in an excel workbook. Further it is not possible to define which sheet should be the default.

**Describe the solution you'd like**
use a query parameter, eg `sheet=` to select the sheet to fetch. in order to make it impossible to fetch sheets that are considered 'private', only sheets with `helix-` or `helix_` prefix should be addressable.
furthermore, if there is a sheet names `helix-default` or `helix_default`, it should be used instead of the first sheet.

/cc @trieloff @davidnuescheler

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #118

### changes

- the sheet inside an excel workbook or google spreadsheet can now be addressed using the `sheet` parameter. 
- only sheets having the `helix-` prefix can be addressed
- if the workbook or spreadsheet does not have any `helix-` prefixed sheets, the first one is returned.
- By default the _used ranged_ of the selected sheet is returned.
- For excel, A table can be addressed using the `table` request parameter, which can be a table name or an index. For example, `table=Table1` will return the table with the name `Table1`, `table=1` will return the second table in the sheet.
- Note that specifying a table index will result in an additional request to lookup the table names.

### CODE REVIEW

**Review by trieloff — CHANGES_REQUESTED** [review]
If we have to break compatibility, let's make sure we update Google and Excel at the same time.

Overall, I'd much prefer a non-breaking change.

**Comment by tripodsan** [comment]
> If we have to break compatibility, let's make sure we update Google and Excel at the same time.

ok.

> Overall, I'd much prefer a non-breaking change.

yes, we could serve the first sheet, if there is no helix-default. but this is one of the security problems we want to fix.
otherwise, you can expose any excel workbook, even the ones not meant for web.

**Comment by trieloff** [comment]
> yes, we could serve the first sheet, if there is no helix-default. but this is one of the security problems we want to fix.
otherwise, you can expose any excel workbook, even the ones not meant for web.

A more gentle change would be to use the new behavior whenever there is a sheet called `helix-*` and serve the first sheet otherwise.

**Comment by tripodsan** [comment]
> > yes, we could serve the first sheet, if there is no helix-default. but this is one of the security problems we want to fix.
> > otherwise, you can expose any excel workbook, even the ones not meant for web.
> 
> A more gentle change would be to use the new behavior whenever there is a sheet called `helix-*` and serve the first sheet otherwise.

exactly not. otherwise you can expose any excel sheet that does **not** have an `helix-*` sheet in it.

**Comment by trieloff** [comment]
You can only expose the first sheet. I wouldn't allow the `table` parameter at all. And this only applies to workbooks that have been shared with Helix in the first place, so I don't see how this would result in additional disclosure.

**Comment by tripodsan** [comment]
> You can only expose the first sheet.

currently, yes. you can only expose the first sheet.
with this enhancement, you can also expose other sheets. as requested in the issue.

> I wouldn't allow the `table` parameter at all. 

that would be another breaking change. in case someone really wants to share a specific table in a sheet, I think this is useful.

> And this only applies to workbooks that have been shared with Helix in the first place, so I don't see how this would result in additional disclosure.

yes - but, people have all sorts of workbooks in their drive that they might expose accidentally. so better safe than sorry.

**Comment by tripodsan** [comment]
> A more gentle change would be to use the new behavior whenever there is a sheet called `helix-*` and serve the first sheet otherwise.

After discussion with @davidnuescheler we decided to go with this idea.

File: README.md [hunk 1]
```diff
@@ -65,6 +65,14 @@ Furthermore, it is possible to limit the result set using `hlx_p.limit` and page
 
 For more, see the [API documentation](docs/API.md).
 
+## Working with Excel and Google Sheets
+
+- The sheet inside an Excel workbook or Google spreadsheet can be addressed using the `sheet` parameter. 
+- Only sheets having the `helix-` prefix can be addressed.
+- If the workbook or spreadsheet does not have any `helix-` prefixed sheets, the first sheet is returned.
+- By default, the _used ranged_ of the selected sheet is returned.
```
Comment (trieloff) [inline README.md:73]:
```suggestion
- By default, the _used range_ of the selected sheet is returned.
```

**Review by trieloff — APPROVED** [review]

**Pull request merged**